### PR TITLE
Test pyupgrade on all files

### PR DESCRIPTION
**Pull request by chris-rands**
Just testing running `pyupgrade --py36-plus` to see if the builds fail. This might save a lot of time over individual PRs but it might be too sweeping and unpleasant to review in one go? Would people be in favour of the `pyupgrade` approach?

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by bow** [comment]
Thanks for trying this out @chris-rands. I would be in favor of at least splitting this into two PRs: one for everything under `Bio` + `BioSQL` and another for everything under `Tests`, in that order.

Seeing the updated code, most of it falls under these categories:
* string formatting (`str.format` and `%`-style formatting replaced with f-strings)
* class definitions without `(object)`
* new-style `super()` statements
* use of `yield from` where appropriate

Aside from the `super()` statements, most of the changes can be evaluated locally by looking at the local context of the statement / expression, which makes them quick to review individually.

Separating the PR for `Tests`, in addition to reducing the load, also ensures that we do not change the tests inadvertently ~ the chance of this happening seems higher with a big, somewhat repetitive changeset ;).

**Comment by peterjc** [comment]
Like Bow, I'd split this up - the ``class Xxxx(object):`` to ``class Xxx:`` for example would make a big but easily reviewed change.

**Comment by peterjc** [comment]
The object change can be done with sed, so maybe start with that?

As to the rest, it like the only way to do this in batches would be to hack https://github.com/asottile/pyupgrade/blob/master/pyupgrade.py to remove lots of fixers...

**Comment by peterjc** [comment]
e.g.

```
$ sed -i.bak -e "s/(object):$/:/g" */*.py Bio/*/*.py Bio/*/*/*.py
```

That seems to work...

**Comment by chris-rands** [comment]
Thanks all- it's possible to limit the string formatting changes of `pyupgrade` via the `--keep-percent-format` flag if preferred?

I can carry on with this since i've started but if we split it up into many PRs it might make good first issues instead?

**Comment by peterjc** [comment]
I suggest you do three pull requests (one of which can be this one updated if you like), in this order:

- objects
- pyupgrade except f-strings
- f-strings

That ought to make 3 manageable ones for review.

**Comment by peterjc** [comment]
Rebased onto master with the object changes via #2453.

File: Bio/Align/Applications/_ClustalOmega.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (peterjc) [inline Bio/Align/Applications/_ClustalOmega.py]: Nice that the tool catches there - Python 3 defaults to UTF8, which is great for using unicode directly in string literals.

**Comment by peterjc** [comment]
With the object changes done, I am hopeful that splitting out the f-string changes will leave a manageable next batch 👍

File: Bio/Align/Applications/_ClustalOmega.py (hunk 1 above)
Comment (peterjc) [reply Bio/Align/Applications/_ClustalOmega.py]: I pull out the five cases of this as #2457 to help break down this monster pull request into more manageable chunks.

**Comment by peterjc** [comment]
Rebasing to current master...

**Comment by peterjc** [comment]
As @bow noted on #2455, some of these exceptions should be ``FileNotFoundError`` which is new in Python 3.

**Comment by peterjc** [comment]
I've ended up doing some ``FileNotFoundError`` fixes in the test on #2469 as part of switching to ``subprocess.getoutput``.

**Comment by chris-rands** [comment]
I propose to close this. After #2480, what is left is the f-strings. If we want to use f-strings instead of `str.format` and decimal formatting etc. across the entire code-base, this will be quite a lot of changes. For example, I ran `flynt` (an alternative to `pyupgrade` focused on f-strings only) on the entire code-case  and it resulted in "255 files changed, 1521 insertions(+), 1672 deletions(-)". So, this might require a module by module approach like the onging efforts with black, if it is worth the errort at all. f-strings are more performant, use fewer characters and can (arguably) be more readable- those are the arguments- thoughts on this?

If we agree to proceed with the goal of adopting f-strings, I will create a new issue and perhaps there will be volunteers to work on it

**Comment by bow** [comment]
> I propose to close this. After #2480, what is left is the f-strings. If we want to use f-strings instead of str.format and decimal formatting etc. across the entire code-base, this will be quite a lot of changes. For example, I ran flynt (an alternative to pyupgrade focused on f-strings only) on the entire code-case and it resulted in "255 files changed, 1521 insertions(+), 1672 deletions(-)". So, this might require a module by module approach like the onging efforts with black, if it is worth the effort at all. f-strings are more performant, use fewer characters and can (arguably) be more readable- those are the arguments- thoughts on this?

> If we agree to proceed with the goal of adopting f-strings, I will create a new issue and perhaps there will be volunteers to work on it

This is a +1 for f-strings from me.

I have been using f-strings almost exclusively these days, falling back to `.format` for some rare cases (mostly when I need a more fine-grained control over when I want to instantiate a template string). In addition to the improved readability, I also find that it can sometimes reduce the need to define variables used only in displayed messages.

**Comment by peterjc** [comment]
OK then, closing as suggested (this is the kind of pull request best merged quickly - thank fully as it is automated, the changes can be regenerated easily).

Let's open a debate about using f-strings - and starting with ``pyupgrade --py36-plus`` would be a sensible first code change.

**Pull request closed without merging**